rest/config.go

```diff
@@ -299,6 +300,43 @@ func (dbConfig DbConfig) validate() error {
 
 }
 
+func (dbConfig *DbConfig) validateSgReaderDbConfig() error {
```

Comment:
In general I think we should move away from the 'reader/writer' terminology.  It was more appropriate when we had a single binary that could run in either mode.  Now that the binaries are split, it's either a Sync Gateway config, or a Sync Gateway Accel config.